pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignment.java

```diff
@@ -23,37 +23,42 @@
 import org.apache.commons.configuration.Configuration;
 import org.apache.helix.HelixManager;
 import org.apache.pinot.common.config.TableConfig;
+import org.apache.pinot.common.utils.InstancePartitionsType;
+import org.apache.pinot.controller.helix.core.assignment.InstancePartitions;
 
 
 /**
- * Strategy to assign segment to instances or rebalance all segments in a table.
+ * Interface for segment assignment and table rebalancing.
```

Comment:
Our documentation states that SegmentAssignmentStrategy is public. You should keep the existing interface and introduce a new one to suit the needs.
https://pinot.readthedocs.io/en/latest/customizations.html#segment-assignment-strategies

Otherwise it is an incompatible change

Reply:
This is not the same interface as the one we mentioned in the documentation, but the new added one for the resource assignment feature. We will deprecate the old interface once the new feature is done, and I will add documentation for the new assignment.